pyxem/utils/io_utils.py

```diff
@@ -460,51 +481,141 @@ def _read_exposures(hdr_info, fp, pct_frames_to_read=0.1, mmap_mode='r'):
         data_type = np.dtype(data_type)
     data_type = data_type.newbyteorder(endian)
 
+    data_mem = np.memmap(fp,
+                         offset=read_offset,
+                         dtype=data_type,
+                         mode=mmap_mode)
+    data_da = da.from_array(data_mem)
+    return data_da
+
+
+def _get_hdr_bits(hdr_info):
+    """
+    gets the number of character bits for the header for each frame given the data type
+    Parameters
+    ----------
+    hdr_info: dict - output of the parse_hdr function
+
+    Returns
+    -------
+    hdr_bits: int - number of characters in the header
+
+    Example
+    -------
+    for 1 bit RAW mib data:
+
+    768
+
+    """
+    data_length = hdr_info['data-length']
+    data_type = hdr_info['data-type']
+
+    if data_type == 'signed':
+        data_type = 'int'
+    elif data_type == 'unsigned':
+        data_type = 'uint'
+    elif data_type == 'float':
+        pass
+    else:
+        raise TypeError('Unknown "data-type" string.')
+
+    # mib data always big-endian
+    endian = '>'
+    data_type += str(int(data_length))
+    # uint1 not a valid dtype
+    if data_type == 'uint1':
+        data_type = 'uint8'
+        data_type = np.dtype(data_type)
+    else:
+        data_type = np.dtype(data_type)
+    data_type = data_type.newbyteorder(endian)
+
     if data_length == '1':
         hdr_multiplier = 1
     else:
-        hdr_multiplier = (int(data_length) / 8)**-1
+        hdr_multiplier = (int(data_length) / 8) ** -1
 
     hdr_bits = int(hdr_info['data offset'] * hdr_multiplier)
 
-    data = np.memmap(fp,
-                     offset=read_offset,
-                     dtype=data_type,
-                     mode=mmap_mode)
-    data = da.from_array(data)
+    return hdr_bits
+
+
+def _read_exposures(hdr_info, fp, pct_frames_to_read=0.1):
+    """
+    Looks into the frame times of the first 10 pct of the frames to see if they are
```

Comment:
This is too specific because 0.1 is a user defined variable - the function should be described generally.